src/main/java/de/vogel612/testclient_javabot/TestProgram.java

```diff
@@ -39,14 +39,23 @@
 
 	public TestProgram() {
 		LOGGER.info("instantiating TestProgram class");
-		AppContext.INSTANCE.add(chatInterface);
-		bot = new DefaultBot(chatInterface);
-		gui = new ClientGui();
-		chatInterface.subscribe(bot);
-		chatInterface.subscribe(gui);
-		LOGGER.info("Basic component setup completed, beginning command glueing.");
-		new CoreBotCommands(chatInterface, bot).allCommands().forEach(bot::subscribe);
-		LOGGER.info("Glued Core Commands");
+        try{
+	        AppContext.INSTANCE.add(chatInterface);
+	        bot = new DefaultBot(chatInterface);
+	        Executors.newSingleThreadExecutor().execute(() -> {
```

Comment:
the `Executors.newSingleThreadExecutor()` created on this line is never shutdown, causing the process to continue running after main window is closed because of a thread that will never die.